### PR TITLE
Tel 55 create patch tenant endpoint

### PR DESCRIPTION
**Pull request by AmitkumarEvry**
Update tenant functionality
Unit test cases

### CODE REVIEW

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsTenantService-CodeBuildProject
* Commit ID: a60d2bf4a62ec3576ed00d90e8f07903bc57ecc6
* Result: FAILED
* [Build Logs](https://tlboygrlvi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e729716c-7cfd-4f3f-9dac-2a07c89528da%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/WWTelematics.TenantService/Application/TenantManager.cs [hunk 1]
```diff
@@ -184,6 +184,71 @@ public ApplicationLoadBalancerResponse CreateTenant(string tenantRequest)
...
+            }
+
+            this.logger.LogTrace("Sending the tenant update request to DB.");
+
+            this.tenantContext.Tenants.Update(tenantUpdate);
+            this.tenantContext.SaveChanges();
+
+            this.logger.LogTrace("Updated the tenant.");
+
+            var responseBody = JsonConvert.SerializeObject(tenantWrapper);
```
Comment (davidw-foxtrax) [inline src/WWTelematics.TenantService/Application/TenantManager.cs]: The response should return the updated object.  This is returning the submission, which is likely incomplete.

File: src/WWTelematics.TenantService/Application/TenantManager.cs [hunk 2]
```diff
@@ -184,6 +184,71 @@ public ApplicationLoadBalancerResponse CreateTenant(string tenantRequest)
...
+            Tenant tenantUpdate = this.tenantContext.Tenants.Where(p => p.PublicId.Equals(publicId)).FirstOrDefault();
+
+            if (tenantUpdate == null)
+            {
+                return ALBResponseHelper.GetDefaultALBResponse(404, "Tenant does not exist.");
+            }
+
+            tenantUpdate.Label = !string.IsNullOrEmpty(tenantWrapper.Tenant.Label) ? tenantWrapper.Tenant.Label : tenantUpdate.Label;
+            tenantUpdate.LicenseCount = tenantWrapper.Tenant.LicenseCount > 0 ? tenantWrapper.Tenant.LicenseCount : tenantUpdate.LicenseCount;
+            tenantUpdate.Active = tenantWrapper.Tenant.Active ? tenantWrapper.Tenant.Active : tenantUpdate.Active;
```
Comment (davidw-foxtrax) [inline src/WWTelematics.TenantService/Application/TenantManager.cs]: This should check whether active is null to determine if the value should be updated.   It should be possible to set Tenant.Active to False.

File: src/WWTelematics.TenantService/Application/TenantManager.cs [hunk 3]
```diff
@@ -184,6 +184,71 @@ public ApplicationLoadBalancerResponse CreateTenant(string tenantRequest)
...
+
+            Tenant tenantUpdate = this.tenantContext.Tenants.Where(p => p.PublicId.Equals(publicId)).FirstOrDefault();
+
+            if (tenantUpdate == null)
+            {
+                return ALBResponseHelper.GetDefaultALBResponse(404, "Tenant does not exist.");
+            }
+
+            tenantUpdate.Label = !string.IsNullOrEmpty(tenantWrapper.Tenant.Label) ? tenantWrapper.Tenant.Label : tenantUpdate.Label;
+            tenantUpdate.LicenseCount = tenantWrapper.Tenant.LicenseCount > 0 ? tenantWrapper.Tenant.LicenseCount : tenantUpdate.LicenseCount;
```
Comment (davidw-foxtrax) [inline src/WWTelematics.TenantService/Application/TenantManager.cs]: The input should be validated before assignment.

Attempting to set LicenseCount to -1 should return an error.

File: src/WWTelematics.TenantService.Tests/Application/TenantManagerTest.cs [hunk 4]
```diff
@@ -309,6 +309,77 @@ public void CanNotReturnMissingTenant()
             Assert.Equal(404, albResponse.StatusCode);
         }
 
+        /// <summary>
+        /// Test the update of a single tenant.
+        /// </summary>
+        [Fact]
+        public void CanUpdateSingleTenant()
```
Comment (davidw-foxtrax) [inline src/WWTelematics.TenantService.Tests/Application/TenantManagerTest.cs]: Can we have a few more tests that check that the requested values were actually updated (for example, testing that licenseCount actually gets changed to the new value but that created_time is not modified?)

File: src/WWTelematics.TenantService/Application/TenantManager.cs (hunk 2 above)
Comment (AmitkumarEvry) [reply src/WWTelematics.TenantService/Application/TenantManager.cs]: @davidw-foxtrax  Tenant.Active is currently not nullable property so if we are not passing active from json request then its value is false. So can we make Active property nullable?
We need to update 3 fields label,license count, and active. So we need to mandatory pass values for these 3 fields in patch request json or need to update single property? 
For example -
**Request json**
{ \"tenant\": { \"label\": \"label update\", \"licenseCount\": 6, \"active\":true  }
or
{ \"tenant\": { \"label\": \"label update\", \"licenseCount\": 6  }
or
{ \"tenant\": { \"label\": \"label update\" }

Comment (davidw-foxtrax) [reply src/WWTelematics.TenantService/Application/TenantManager.cs]: * The Patch endpoint must be able to accept partial objects.  Active is never required.
* The Patch endpoint must be able to change Active to false.

If making the field nullable is the way to solve this, we'll do that.  But, make sure this doesn't impact that

* Active is a required field when saving data to the db

**Comment by davidw-foxtrax** [comment]
Replaced by #12

**Pull request closed without merging**